### ISSUE TITLE
ConfigWatch requires spring-boot-actuator

### ISSUE THREAD

**Issue by msvab** [spring-cloud/spring-cloud-consul#274]
Hi,

I'm using consul config in a spring application that doesn't have a web server. So I didn't include spring-boot-actuator, since I didn't see the need for it. However in `ConsulConfigAutoConfiguration` the `ConfigWatch` is enabled only when `RefreshEndpoint` from actuator is available.

If I put the following into my app, ConfigWatch works as expected without the actuator dependency.
```
@EnableScheduling
public class App extends SpringApplication {

    @Bean
    public ConfigWatch configWatch(ConsulConfigProperties properties, ConsulPropertySourceLocator locator, ConsulClient consul) {
        return new ConfigWatch(properties, locator.getContexts(), consul);
    }
}
```

Actuator enables scheduling automatically in `MetricExportAutoConfiguration` and so it would have to be enabled in `ConsulConfigAutoConfiguration`. I don't think there would be any side-effects from this change.

Cheers,
Michal

**Comment by spencergibb** [comment]
So the `ConfigWatch` may not require actuator, but the `RefreshEventListener` does require it.

**Comment by msvab** [comment]
You're right, I've missed that! The `RefreshEventListener` call `RefreshEndpoint` only to call `ContextRefresher.refresh()`. Would it be possible to inject `ContextRefresher` directly into `RefreshEventListener` instead of `RefreshEndpoint`?

**Comment by spencergibb** [comment]
Of course, anything is possible, but I'd rather not introduce that coupling. The right thing to do is install the `RefreshEventListener` regardless of if actuator is present.

Depends on https://github.com/spring-cloud/spring-cloud-commons/issues/171

**Comment by msvab** [comment]
Alright, sounds good to me. Thanks!
